src/main/java/uk/gov/justice/hmpps/casenotes/controllers/SyncController.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.hmpps.casenotes.controllers
+
+import io.swagger.v3.oas.annotations.tags.Tag
+import jakarta.validation.Valid
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.hmpps.casenotes.config.SecurityUserContext.Companion.ROLE_CASE_NOTES_SYNC
+import uk.gov.justice.hmpps.casenotes.sync.SyncCaseNoteRequest
+import uk.gov.justice.hmpps.casenotes.sync.SyncCaseNotes
+import uk.gov.justice.hmpps.casenotes.sync.SyncResult
+
+@Tag(name = "Sync Case Notes")
+@RestController
+@RequestMapping("sync/case-notes")
+class SyncController(private val sync: SyncCaseNotes) {
+  @PutMapping
+  @PreAuthorize("hasRole('$ROLE_CASE_NOTES_SYNC')")
+  fun syncCaseNotes(@Valid @RequestBody caseNotes: List<SyncCaseNoteRequest>): List<SyncResult> = sync.caseNotes(caseNotes)
```

Comment:
Probably worth adding the standard 4xx responses here, maybe in another PR

Reply:
At some point, yes. It's a well used API so we should try and document it as well as possible. It's definitely somewhat lacking doc wise at the mo